# SPDX-License-Identifier: Apache-2.0
# Upstream libcupsfilters e14bad406189a66bdeb40131112944bfb58ee0cc ("Fix printing error: pdfio
# output "Missing Root object."", #167), unreleased after 2.2.1. cfPDFPagesFP() copies its input
# into a temporary file and opens it with pdfio without flushing stdio first, so raster jobs fail
# (0.0.4.1 K2, D84). Drop with the override once FDSDK ships a release containing it.

diff --git a/cupsfilters/pdf.c b/cupsfilters/pdf.c
index 1f714b203..3e4386091 100644
--- a/cupsfilters/pdf.c
+++ b/cupsfilters/pdf.c
@@ -165,6 +165,11 @@ cfPDFPagesFP(FILE *file) 	// I - File pointer of PDF
     fwrite(buffer, 1, bytes_read, temp_fp);
   }
 
+  // call fflush(temp_fp) to flush the stdio buffer; 
+  // otherwise pdfioFileOpen() may not see the complete contents of temp_filename because 
+  // some data may still be buffered and not yet written to the file.
+  fflush(temp_fp);  
+
   // Open the temp file with pdfio
   pdfio_file_t *pdf = pdfioFileOpen(temp_filename, NULL, NULL, NULL, NULL);
 
